File: TopCoder/Sorting/SRM606_.py
class EllysSubstringSorter:
 def getMin(self, S, L):
     
    # generate device output for i
    def device(i):
        return S[0:i] + ( ''.join( sorted(S[i:i+L]) ) ) + S[i+L:]
      
    # pick minimum (lexicographically-first) possible output:
    return min( device(i) for i in range( len(S) - L + 1) )

File: TopCoder/Sorting/SRM606.cpp
//Div 2-250
#include <string>
#include <iostream>
#include <algorithm>
using namespace std;
class EllysSubstringSorter{
public:
	string getMin(string S, int L){
		int len = S.length() - L;
		string result(S);
		for (int i = 1; i <= len; ++i){
			string t = S;
			sort(t.begin() + i, t.begin() + i + L);
			//the point 字典序比较
			result = min(result, t);
		}
		return result;
	}
};
int main()
{
	EllysSubstringSorter ss;
	cout << ss.getMin("ESPRIT", 3);
}

